src/main/java/step3/Car.java

```diff
@@ -1,13 +1,23 @@
 package step3;
 
 public class Car {
-    private int trace = 0;
+    private final String name;
+    private int trace;
 
-    public void plusTrace() {
-        this.trace += 1;
+    public Car(String name) {
+        this.name = name;
+        this.trace = 0;
+    }
+
+    public String getName() {
+        return this.name;
     }
 
     public int getTrace() {
         return this.trace;
     }
+
+    public void plusTrace() {
```

Comment:
이동 여부를 판단하는 로직을 Car가 담당하는 것은 어떨까?

```suggestion
    public void move(int randomNo) {
        if (randomNo >= 4) {
             this.trace += 1;
        }
```

Reply:
이동 여부 판단할 로직은 Car가 책임을 맡는게 도메인 이해하기가 쉬워보이네요~ 감사합니다!